src/components/interaction-designer/flow-editors/import/LanguageAdder.vue

```diff
@@ -0,0 +1,253 @@
+            :options="iso_639_3Tags"
+            :allow-empty="false"
+            :custom-label="customLanguageLabel"
+            track-by="iso6393"
+            :searchable="true" />
+        </validation-message>
+      </div>
+      <div class="form-group">
+        <validation-message
+          #input-control="{ isValid }"
+          message-key="language/new_language/variant">
+          <text-editor
+            v-model="variant"
+            :placeholder="'flow-builder.enter-variant' | trans"
+            :label="'flow-builder.language-variant' | trans"
+            :valid-state="isValid"
+            @keydown="filterVariant" />
+        </validation-message>
+      </div>
+      <div class="form-group">
+        <label class="form-check-label mt-2 mb-2 mr-2">{{'flow-builder.iso-3166-1-label' | trans}}</label>
+        <vue-multiselect
+          v-model="iso_3166_1"
+          :placeholder="'flow-builder.language-locale-selector-placeholder' | trans"
+          :options="iso_3166_1Locales()"
+          :allow-empty="false"
+          track-by="locale"
+          :custom-label="customLocaleLabel"
+          :searchable="true" />
+      </div>
+      <div class="form-group">
+        <label class="form-check-label mt-2 mb-2 mr-2">{{'flow-builder.bcp-47-label' | trans}}</label>
+        <validation-message
+          #input-control="{ isValid }"
+          message-key="language/new_language/bcp_47">
+          <input
+            v-model="newLanguage.bcp_47"
+            name="bcp_47"
+            type="text"
+            class="form-control"
+            disabled>
+        </validation-message>
+      </div>
+    </b-modal>
+  </div>
+</template>
+
+<script lang="ts">
+
+import {BModal} from 'bootstrap-vue'
+import Lang from '@/lib/filters/lang'
+import Routes from '@/lib/mixins/Routes'
+import Vue from 'vue'
+import {Component} from 'vue-property-decorator'
+import {mixins} from 'vue-class-component'
+import {iso6393} from 'iso-639-3'
+
+import {
+  map,
+  keys,
+  isEmpty,
+} from 'lodash'
+
+import {
+  ILanguage,
+  IContext,
+} from '@floip/flow-runner'
+import {Action, namespace, State} from 'vuex-class'
+
+import {IdGeneratorUuidV4} from '@floip/flow-runner/dist/domain/IdGeneratorUuidV4'
+
+import ValidationMessage from '@/components/common/ValidationMessage.vue'
+import TextEditor from '@/components/common/TextEditor.vue'
+import VueMultiselect from 'vue-multiselect'
+
+const countries = require('i18n-iso-countries')
+countries.registerLocale(require('i18n-iso-countries/langs/en.json'))
+countries.registerLocale(require('i18n-iso-countries/langs/fr.json'))
+
+const importVuexNamespace = namespace('flow/import')
+const validationVuexNamespace = namespace('validation')
+
+@Component({
+  components: {
+    BModal,
+    ValidationMessage,
+    VueMultiselect,
+    TextEditor,
+  },
+})
+class LanguageAdder extends mixins(Lang, Routes) {
+  addLanguageError = ''
+  newLanguage = {
```

Comment:
I think using the `ILanguage` interface to initiate an object would be cleaner:
`newLanguage = {} as ILanguage`